docs/instructor-training-workflow.txt

```diff
@@ -0,0 +1,211 @@
+## Notation
+
+- [use cases] are in square brackets
+- {web pages} are in curly braces
+- <database objects> are in angle brackets
+
+## Individual application for training.
+
+* [user registers self][a]: user match by email address or GitHub ID.
+  - User fills in {single user application form} on AMY with name, email address, GitHub ID, etc.
+  - AMY matches to pre-existing <Person> using email or GitHub ID.
+  - <Training Application> created, connected to pre-existing <Person>, marked "Pending", and added to <Training Application List>.
+
+* [user registers self][b]: no user match.
+  - User fills in form on AMY with name, email address, GitHub ID, etc.
+  - AMY does *not* match email or GitHub ID.
+  - <Training Application> created, marked "Pending", and added to <Training Application List> *without* <Person> object.
+
+## Administrator approves individual for training.
+
+* [admin approves training][a]: user record already in database.
+  - Admin views <Training Application List>.
+  - Admin selects <Training Application> marked "Pending" that has been matched with <Person>.
+  - Admin approves <Training Application>.
+  - AMY clears <Training Application>'s "Pending" flag.
+
+* [admin approves training][b]: no user record in database.
+  - Admin views <Training Application List>.
+  - Admin selects <Training Application>.
+  - Admin approves <Training Application>.
+  - AMY clears <Training Application>'s "Pending" flag.
+  - AMY creates <Person> record and associates with <Training Application>.
```

Comment:
1) Do you want `may_contact` set to True? If yes, should we add a note to the `TrainingRequestForm` that we may contact them?
2) What is the password for the new Person record?
3) Is there any reason for the "Pending" state except that we need to match application to a Person?
4) What if somebody makes two requests? The admin probably needs something like "discard this application".

Do I understand correctly that:

- We use `create_username` and ignore GitHub login when choosing applicant username. 
- We set `airport` to null, because we don't have such field in the form.
- In the new Person record, we leave `twitter`, `url`, `notes`, `badges`, `lessons`, `domains`, `languages` and `orcid` empty.
- After approving the request, the applicant still cannot login using GitHub and `Person.is_active == False`.